src/main/java/com/google/opensesame/servlets/MentorObject.java

```diff
@@ -1,19 +1,43 @@
 package com.google.opensesame.servlets;
 
-public class MentorObject {
-  private String name;
-  private String description;
+import java.util.ArrayList;
 
-  public MentorObject(String nameInput, String descriptionInput) {
-    name = nameInput;
-    description = descriptionInput;
+public class MentorObject extends PersonObject {
+  private ArrayList<String> projectIDs;
+
+  public MentorObject(String nameInput, String gitHubIDInput) {
+    this(nameInput, gitHubIDInput, "", new ArrayList<String>(), new ArrayList<String>());
+  }
+
+  public MentorObject(String nameInput, String gitHubIDInput, String descriptionInput) {
+    this(
+        nameInput,
+        gitHubIDInput,
+        descriptionInput,
+        new ArrayList<String>(),
+        new ArrayList<String>());
+  }
+
+  public MentorObject(
+      String nameInput, String gitHubIDInput, String descriptionInput, ArrayList<String> tagInput) {
+    this(nameInput, gitHubIDInput, descriptionInput, tagInput, new ArrayList<String>());
+  }
+
+  public MentorObject(
+      String nameInput,
+      String gitHubIDInput,
+      String descriptionInput,
+      ArrayList<String> tagInput,
+      ArrayList<String> projectIDInputs) {
+    super(nameInput, gitHubIDInput, descriptionInput, tagInput);
+    projectIDs = projectIDInputs;
   }
```

Comment:
This is a lot of constructors, do you think we could achieve the same effect by passing `null` to various constructor arguments and handling those accordingly? For example:
```java
public MentorObject(
      String nameInput,
      String gitHubIDInput,
      String descriptionInput,
      ArrayList<String> tagInput,
      ArrayList<String> projectIDInputs) {
    if (tagInput == null) {
      tagInput = new ArrayList<String>();
    }
    [...]
    super(nameInput, gitHubIDInput, descriptionInput, tagInput);
    projectIDs = projectIDInputs;
  }
```

Open to discussion on this.

Reply:
Also, something else that I am now thinking of after having looked at the `PersonObject` class is that both the `MentorObject` and future `MenteeObject` class could potentially be more cleanly implemented if the `PersonObject` was stored as a property within these classes instead of as their parent class. Then the `PersonObject` could be passed through the constructor instead of having to pass in every individual property required to construct the `PersonObject`. Is there something you were specifically looking to implement by taking the inheritance approach?

Reply:
One other thing to consider is that it seems like Gson has some trouble working with polymorphism out of the box. It would most likely require some extra configuration on our end to get serialization to work: https://stackoverflow.com/questions/5800433/polymorphism-with-gson

Reply:
I agree with Richie that this is a lot of constructors. I think that the builder pattern would work well here. Here is a good stackoverflow post describing the builder pattern: https://stackoverflow.com/questions/222214/managing-constructors-with-many-parameters-in-java/222295#222295.

Reply:
@Richie78321 I believe after our Pod Sunk discussion yesterday, we are now all on the same page about inheritance and class structure.

Reply:
Thank you Joe! I will try out that approach.